src/Evolution/Systems/ScalarWave/Characteristics.hpp

```diff
@@ -0,0 +1,225 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Evolution/Systems/ScalarWave/Tags.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <typename>
+class Variables;
```

Comment:
You don't have a `Variables` anywhere

Reply:
I do now, after replacing occurrences of `typename Tags::CharacteristicSpeeds<Dim>::type` and similar with their explicit types